crates/sui-core/src/traffic_controller.rs

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::net::SocketAddr;
+use std::{collections::HashMap, sync::Arc};
+
+use mysten_metrics::spawn_monitored_task;
+use parking_lot::RwLock;
+use std::time::{Duration, SystemTime};
+use sui_types::traffic_control::{
+    Policy, PolicyConfig, PolicyResponse, TrafficControlPolicy, TrafficTally,
+};
+use tokio::sync::mpsc;
+use tokio::sync::mpsc::error::TrySendError;
+use tracing::warn;
+
+type BlocklistT = Arc<RwLock<HashMap<SocketAddr, SystemTime>>>;
```

Comment:
I think it will actually simplify the code and be faster to just use DashMap, then you don't need all the `.read()` and `.write()` calls.

Another option is MokaCache, which would let us easily bound the size of the block list to prevent memory attacks. There are examples of its usage in `writeback_cache.rs`